android/src/main/java/com/mapbox/mapboxgl/MapboxMapController.java

```diff
@@ -1403,6 +1414,12 @@ public void setMyLocationRenderMode(int myLocationRenderMode) {
     }
   }
 
+  @Override
+  public void setMyLocationStyle(Map myLocationStyle) {
+    this.myLocationStyle = myLocationStyle;
+    updateLocationComponentLayer();
```

Comment:
@felix-ht can you take a look?
`updateLocationComponentLayer()` cannot refresh new color after setState or hot reload